tests/memory.x

```diff
@@ -1,6 +1,7 @@
 MEMORY
 {
-    FLASH      (rx)  :  ORIGIN = 0x10000000,                             LENGTH = 512K
+    FLASH      (rx)  :  ORIGIN = 0x1000E000,                             LENGTH = 222K 
```

Comment:
We shouldn't actually change the memory.x in the tests to add SECFLASH and change the FLASH. Keep FLASH where it is and at 512K - these need to work without a MITRE bootloader

Remove the SECFLASH stuff and the _stext setting too